segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaConnectionListener.java

```diff
@@ -173,19 +205,73 @@ public void initChannel(SocketChannel ch) throws Exception {
              }
          });
 
+        if (enableTls && enableTlsReload) {
+            log.info("TLS reload is enabled, so setting up a FileModificationMonitor object to monitor/watch changes in file: [{}]",
+                    this.pathToTlsCertFile);
+
+            tlsCertFileModificationMonitor = prepareCertificateMonitor(this.pathToTlsCertFile, this.pathToTlsKeyFile,
+                    sslCtx);
+            tlsCertFileModificationMonitor.startMonitoring();
+            log.info("Started the FileModificationMonitor object successfully");
+        }
+
         // Start the server.
         serverChannel = b.bind(host, port).awaitUninterruptibly().channel();
     }
 
+    @VisibleForTesting
+    FileModificationMonitor prepareCertificateMonitor(String tlsCertificatePath, String tlsKeyPath,
+                                                      AtomicReference<SslContext> sslCtx) {
+        return prepareCertificateMonitor(Files.isSymbolicLink(Paths.get(tlsCertificatePath)),
+                tlsCertificatePath, tlsKeyPath, sslCtx);
+
+    }
+
+    @VisibleForTesting
+    FileModificationMonitor prepareCertificateMonitor(boolean isTLSCertPathSymLink, String tlsCertificatePath,
+                                                      String tlsKeyPath,
+                                                      AtomicReference<SslContext> sslCtx) {
```

Comment:
Why is this AtomicReference? You do not need this.

Reply:
The SSL Context gets reloaded via the callback invoked by file modification event watcher or polling monitor. The SSL context might be in use while its reload gets triggered by the callback.